auth/tracing/groups.go

```diff
@@ -41,89 +41,100 @@ func GroupRepositoryMiddleware(tracer opentracing.Tracer, gr auth.GroupRepositor
 
 	return grm.repo.RetrieveByID(ctx, id)
 }
 
 func (grm groupRepositoryMiddleware) RetrieveAllParents(ctx context.Context, groupID string, pm auth.PageMetadata) (auth.GroupPage, error) {
-	span := createSpan(ctx, grm.tracer, retrieveAllParents)
+	span := grm.createSpan(ctx, grm.tracer, retrieveAllParents)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.RetrieveAllParents(ctx, groupID, pm)
 }
 
 func (grm groupRepositoryMiddleware) RetrieveAllChildren(ctx context.Context, groupID string, pm auth.PageMetadata) (auth.GroupPage, error) {
-	span := createSpan(ctx, grm.tracer, retrieveAllChildren)
+	span := grm.createSpan(ctx, grm.tracer, retrieveAllChildren)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.RetrieveAllChildren(ctx, groupID, pm)
 }
 
 func (grm groupRepositoryMiddleware) RetrieveAll(ctx context.Context, pm auth.PageMetadata) (auth.GroupPage, error) {
-	span := createSpan(ctx, grm.tracer, retrieveAll)
+	span := grm.createSpan(ctx, grm.tracer, retrieveAll)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.RetrieveAll(ctx, pm)
 }
 
 func (grm groupRepositoryMiddleware) Memberships(ctx context.Context, memberID string, pm auth.PageMetadata) (auth.GroupPage, error) {
-	span := createSpan(ctx, grm.tracer, memberships)
+	span := grm.createSpan(ctx, grm.tracer, memberships)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.Memberships(ctx, memberID, pm)
 }
 
 func (grm groupRepositoryMiddleware) Members(ctx context.Context, groupID, groupType string, pm auth.PageMetadata) (auth.MemberPage, error) {
-	span := createSpan(ctx, grm.tracer, members)
+	span := grm.createSpan(ctx, grm.tracer, members)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.Members(ctx, groupID, groupType, pm)
 }
 
 func (grm groupRepositoryMiddleware) Assign(ctx context.Context, groupID, groupType string, memberIDs ...string) error {
-	span := createSpan(ctx, grm.tracer, assign)
+	span := grm.createSpan(ctx, grm.tracer, assign)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.Assign(ctx, groupID, groupType, memberIDs...)
 }
 
 func (grm groupRepositoryMiddleware) Unassign(ctx context.Context, groupID string, memberIDs ...string) error {
-	span := createSpan(ctx, grm.tracer, unassign)
+	span := grm.createSpan(ctx, grm.tracer, unassign)
 	defer span.Finish()
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
 	return grm.repo.Unassign(ctx, groupID, memberIDs...)
 }
+
+func (grm groupRepositoryMiddleware) createSpan(ctx context.Context, tracer opentracing.Tracer, opName string) opentracing.Span {
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
+		return tracer.StartSpan(
+			opName,
+			opentracing.ChildOf(parentSpan.Context()),
+		)
+	}
+
+	return tracer.StartSpan(opName)
+}
```

Comment:
```suggestion
func (grm groupRepositoryMiddleware) createSpan(ctx context.Context, tracer opentracing.Tracer, opName string) (opentracing.Span, context.Context) {
	return opentracing.StartSpanFromContextWithTracer(ctx, tracer, opName)
}
```

Reply:
since it's a single line you can even consider removing the create span function

Reply:
yes I can just place `opentracing.StartSpanFromContextWithTracer(ctx, tracer, opName)` directly where we call the create span function